unittest/joint-configurations.cpp

```diff
@@ -282,7 +282,7 @@ BOOST_AUTO_TEST_CASE ( integrate_difference_test )
  Eigen::VectorXd q1(randomConfiguration(model, -1 * Eigen::VectorXd::Ones(model.nq), Eigen::VectorXd::Ones(model.nq) ));
  Eigen::VectorXd qdot(Eigen::VectorXd::Random(model.nv));
 
- BOOST_CHECK_MESSAGE(isSameConfiguration(model, integrate(model, q0, difference(model, q0,q1)), q1), "Integrate (difference) - wrong results");
+ BOOST_CHECK_MESSAGE(isSameConfiguration(model, integrate(model, q0, difference(model, q0,q1)), q1, 1.5e-12), "Integrate (difference) - wrong results");
```

Comment:
Why did you change the prec here?

Reply:
With the new diff_impl, `integrate(q0, difference(q0,q1))` is less similar to the expected result `q1`. I am not totally sure why this is the case, but probably because the current
implementation of integrate also suffers from floating point error... The threshold for the integrate_difference_test needs to be changed slightly to accommodate the new implementation of difference.

Reply:
It is highly surprising and this needs additional investigation for this PR to be merged. Could you look at it?

Reply:
I will need help for this since I am not too familiar with the mathematics :sweat_smile: For this issue I actually found the alternative implementation thanks to your `// TODO: perform first the Quaternion multiplications and then return a Rotation Matrix` note under `dDifference_impl` in `special-orthogonal.hpp`.

Reply:
Could you see with @jmirabel? I guess you are already in contact with him.